pkg/odo/cli/init/init.go

```diff
@@ -175,6 +175,18 @@ func (o *InitOptions) Run() (err error) {
 	if err = devfileObj.SetMetadataName(name); err != nil {
 		return err
 	}
+	scontext.SetComponentType(o.ctx, component.GetComponentTypeFromDevfileMetadata(devfileObj.Data.GetMetadata()))
+	scontext.SetLanguage(o.ctx, devfileObj.Data.GetMetadata().Language)
+	scontext.SetProjectType(o.ctx, devfileObj.Data.GetMetadata().ProjectType)
+
+	// Set the name in the devfile *AND* writes the devfile back to the disk in case
+	// it has been removed and not replaced by the starter project
+	_, err = o.clientset.InitClient.PersonalizeName(devfileObj, o.flags)
+	if err != nil {
+		return fmt.Errorf("failed to update the devfile's name: %w", err)
+	}
```

Comment:
Why are you adding this code? rebasing error?

Reply:
I would guess should it be removed?

Reply:
done